### PR TITLE
Add support for managed MySQL Flexible server

### PR DESCRIPTION
**Pull request by yash-nisar**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yash-nisar** [comment]
It takes ~6mins to provision a flexible mysql server from the command line so I don't think it's feasible to add ITs: 
image

**Comment by yash-nisar** [comment]
Verified the binding works: 
image

**Comment by yash-nisar** [comment]
Also could see env vars on curling the app: 
```
AZURE_MYSQL_DATABASE=XX
AZURE_MYSQL_HOST=XX
AZURE_MYSQL_PASSWORD=XX
AZURE_MYSQL_PORT=XX
AZURE_MYSQL_SSL=XX
AZURE_MYSQL_USERNAME=XX
```
Command to bind: 
```
az containerapp update -n appName -g rgName --bind mysqlName:mysqlconnName,username=XX,password=XX,database=XX
```

File: src/containerapp/azext_containerapp/_managed_service_utils.py [hunk 1]
```diff
@@ -126,8 +126,56 @@ def build_postgresql_service_connector_def(subscription_id, resource_group_name,
...
                 "Managed PostgreSQL Flexible Server needs the database, username, and password arguments.")
-        resource_id = ManagedPostgreSQLUtils.build_postgres_resource_id(subscription_id, resource_group_name,
+        resource_id = ManagedPostgreSQLFlexibleUtils.build_postgres_resource_id(subscription_id, resource_group_name,
+                                                                                service_name, arg_dict)
+        parameters = ManagedPostgreSQLFlexibleUtils.build_postgres_params(resource_id, name, arg_dict["username"],
+                                                                          arg_dict["password"], key_vault_id=None)
+        return {"linker_name": binding_name, "parameters": parameters, "resource_id": resource_id}
+
+
+class ManagedMySQLFlexibleUtils:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_managed_service_utils.py:136]: Please add tests for
az containerapp create/update --bind

**Comment by zhoxing-ms** [comment]
Could you please add some test case for this PR change?

**Comment by yash-nisar** [comment]
@bgashirabake is working on the tests and will continue this PR.

**Comment by bgashirabake** [comment]
> Could you please add some test case for this PR change?

done. could you please take a look when you get the chance. thank you!!

**Review by Greedygre — APPROVED** [review]

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py [hunk 2]
```diff
@@ -767,6 +768,43 @@ def test_containerapp_dev_service_binding_e2e(self, resource_group):
         self.cmd('containerapp service list -g {} --environment {}'.format(resource_group, env_name), checks=[
             JMESPathCheck('length(@)', 0),
         ])
+    @AllowLargeResponse(8192)
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:771]: Blank lines are required between each test

```suggestion

    @AllowLargeResponse(8192)
```

Comment (bgashirabake) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_commands.py:772]: committed suggestion and updated, thank you!

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms  @yanzhudd 
Could you help to review and merge this PR? Thanks.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**